include/res.h

```diff
@@ -45,14 +46,14 @@
 /*! \brief An item. */
 struct s_item
 {
-    /*! Name of the item (16 chars plus '\0') */
-    char name[17];
+    /*! Item's short name (around 16 characters). */
+    std::string name;
```

Comment:
I looked as carefully as possible that there's no C memory management going on with `s_item` (like memcpy, memmove, strcpy*, strlen, etc.), but I may have overlooked a few.

If we suddenly get strange crashes due to strings, they almost never manifest at the place of the *actual* error; they tend to abort when something that seems unrelated goes out of scope and it causes a double-delete or something.

\* The C version of `strcpy()`, not the `std::string` overloaded one that I added in earlier.

Reply:
Got an error relating to this:
```
In file included from /Projects/kq-fork/include/kq.h:47:
/Projects/kq-fork/include/utilities.h:50:60: error: cannot pass object of non-trivial type 'std::__1::basic_string<char>' through variadic function; call will abort at runtime [-Wnon-pod-varargs]
    int size_s = std::snprintf(nullptr, 0, format.c_str(), args...);
                                                           ^
/Projects/kq-fork/src/combat.cpp:1322:21: note: in instantiation of function template specialization 'sprintf<std::__1::basic_string<char> >' requested here
                    sprintf(strbuf, _("%s found!"), items[found_item].item_name);
                    ^
```
I think possibly needs to be `items[found_item].item_name.c_str()` ?

Reply:
Also
* hskill.cpp:794
* intrface.cpp:1435
* intrface.cpp:1439
* intrface.cpp:1451
* intrface.cpp:1455

Reply:
Looking into this now. Frustrated that my compiler gave zero warnings about this.

Reply:
The latest push should take care of this.

And in doing so, I noticed that `s_item::item_name` was changed to a std::string, but `s_spell::name` was not. I'll probably want to do that...